pkg/util/sys/device.go

```diff
@@ -42,6 +41,44 @@ type Partition struct {
 	Label string
 }
 
+// LocalDevice contains information about an unformatted block device
+type LocalDisk struct {
+	// Name is the device name
+	Name string `json:"name"`
+	// Parent is the device parent's name
+	Parent string `json:"parent"`
+	// HasChildren is whether the device has a children device
+	HasChildren bool `json:"hasChildren"`
+	// DevLinks is the persistent device path on the host
+	DevLinks string `json:"devLinks"`
+	// Size is the device capacity in byte
+	Size uint64 `json:"size"`
+	// UUID is used by /dev/disk/by-uuid
+	UUID string `json:"uuid"`
+	// Serial is the disk serial used by /dev/disk/by-id
+	Serial string `json:"serial"`
+	// Type is disk type
+	Type string `json:"type"`
+	// Rotational is the boolean whether the device is rotational: true for hdd, false for ssd and nvme
+	Rotational bool `json:"rotational"`
+	// ReadOnly is the boolean whether the device is readonly
+	Readonly bool `json:"readOnly"`
+	// OwnPartition is whether rook owns the partition
+	OwnPartition bool `json:"ownPartition"`
```

Comment:
when i tested the branch, `OwnPartition` was always false even on devices where the osd was configured with bluestore partitions.

Reply:
i see, `blkid` needs to run in privileged mode to read partition label.

Reply:
i'll use udevadm to read parititon label

Reply:
sample output
```
[root@rook-discover-lc57t /]# blkid /dev/sdc3 -s PARTLABEL -o value
[root@rook-discover-lc57t /]# udevadm info /dev/sdc3 |grep ROOK    
S: disk/by-partlabel/ROOK-OSD2-BLOCK
E: DEVLINKS=/dev/disk/by-id/scsi-3600140568c0bd28d4ee43769387c9f02-part3 /dev/disk/by-id/wwn-0x600140568c0bd28d4ee43769387c9f02-part3 /dev/disk/by-partlabel/ROOK-OSD2-BLOCK /dev/disk/by-partuuid/2089640e-bdeb-4fb4-aaec-88e165780b88 /dev/disk/by-path/ip-127.0.0.1:3260-iscsi-iqn.2016-06.world.srv:storage.target01-lun-2-part3
E: ID_PART_ENTRY_NAME=ROOK-OSD2-BLOCK
E: PARTNAME=ROOK-OSD2-BLOCK
```